src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -70,13 +72,40 @@ case object UnknownPeriod extends Period {
 }
 
 case class PeriodSum(periods: Set[Period], modifier: Modifier) extends Period {
-  override def addTo(temporal: Temporal): Temporal = ???
 
-  override def get(unit: TemporalUnit): Long = ???
+  var map = scala.collection.mutable.Map[TemporalUnit,Long]()
 
-  override def subtractFrom(temporal: Temporal): Temporal = ???
+  for ( period <- periods; unit <- period.getUnits.asScala )
+    map.get(unit) match {
+      case Some(value) => map.update( unit, value + period.get(unit))
```

Comment:
Unfortunately, that was my initial attempt, but I had to use the match keyword because there is nothing present in the map when it is instantiated, the map throws a NoSuchElementException because the key that we are trying to access has not been added yet. 

Or do you mean line 80 specifically?